src/pkg/packaging-tools/framework.dependency.targets

```diff
@@ -53,6 +56,100 @@
+    </PropertyGroup>
+    <ItemGroup>
+      <CoreCLRFiles Include="$(CoreCLROverridePath)/*.*" />
+      <CoreCLRFiles Include="$(CoreCLROverridePath)/Redist/**/*.dll" />
+      <CoreCLRFiles Condition="'$(CoreCLRPDBOverridePath)' != ''" Include="$(CoreCLRPDBOverridePath)/*.pdb;$(CoreCLRPDBOverridePath)/*.dbg" />
+
+      <OverriddenRuntimeFiles Include="@(ReferenceCopyLocalPaths)" Condition="'@(CoreCLRFiles->'%(FileName)%(Extension)')' == '%(FileName)%(Extension)'">
+        <CoreCLRFile>@(CoreCLRFiles)</CoreCLRFile>
+      </OverriddenRuntimeFiles>
+
+      <LongNameDacFile Include="@(ReferenceCopyLocalPaths)" Condition="$([System.String]::Copy('%(FileName)').StartsWith('mscordaccore_'))">
+        <TargetPath>%(PathInPackage)</TargetPath>
+      </LongNameDacFile>
+
+      <ShortNameDacFileOverride Include="@(CoreCLRFiles)" Condition="'%(FileName)%(Extension)' == 'mscordaccore.dll'" />
+
+      <OverriddenRuntimeFiles Include="@(LongNameDacFile)">
+        <CoreCLRFile>@(ShortNameDacFileOverride)</CoreCLRFile>
+      </OverriddenRuntimeFiles>
+
+      <OverriddenRuntimeFiles>
+        <IsNative Condition="$([System.String]::new('%(Identity)').ToLowerInvariant().Replace('\', '/').Contains('/native/'))">true</IsNative>
+      </OverriddenRuntimeFiles>
+
+      <ReferenceCopyLocalPaths Remove="@(OverriddenRuntimeFiles)" />
+      <ReferenceCopyLocalPaths Include="@(OverriddenRuntimeFiles->Metadata('CoreCLRFile'))" />
+      
+    </ItemGroup>
+
+    <Error Condition="'@(CoreCLRFiles)' == ''" Text="The path provided to CoreCLROverridePath ($(CoreCLROverridePath)) does not contain any files." />
+  </Target>
+
+  <PropertyGroup>
+    <CoreFXSharedFrameworkPackageName>Microsoft.Private.CoreFx.NETCoreApp</CoreFXSharedFrameworkPackageName>
+  </PropertyGroup>
+
+  <Target Name="OverrideFrameworkFilesFromPackageResolve" Condition="'$(CoreFXOverridePath)' != ''" BeforeTargets="GetFilesFromPackageResolve">
+    <Error Condition="!Exists('$(CoreFXOverridePath)')" Text="The path provided to CoreFXOverridePath ($(CoreFXOverridePath)) does not exist." />
+
+    <PropertyGroup>
+      <CoreFXSharedFrameworkPackageSpec>$(CoreFXOverridePath)/specs/$(CoreFXSharedFrameworkPackageName).nuspec</CoreFXSharedFrameworkPackageSpec>
+      <CoreFXSharedFrameworkImplementationPackageSpec Condition="'$(PackageRID)' != ''">$(CoreFXOverridePath)/specs/runtime.$(PackageRID).$(CoreFXSharedFrameworkPackageName).nuspec</CoreFXSharedFrameworkImplementationPackageSpec>
+    </PropertyGroup>
+
+    
+    <Error Condition="!Exists('$(CoreFXSharedFrameworkPackageSpec)')" Text="The nuspec for the reference Microsoft.Private.CoreFx.NETCoreApp package does not exist." />
+    <Message Condition="'$(CoreFXSharedFrameworkImplementationPackageSpec)' != '' And !Exists('$(CoreFXSharedFrameworkImplementationPackageSpec)')"
+      Text="The nuspec for the implementation runtime.$(PackageRID).Microsoft.Private.CoreFx.NETCoreApp package does not exist. Falling back to packaged CoreFX." />
```

Comment:
When is it desirable to let this happen? Would it be better to error out to catch the maybe more common case of the CoreFX build failing to produce it, but allow passing another property if it's truly what you want?

Should the message have `Importance="High"` so the user has a chance of seeing it in the logs?

(All depends on scenario, I guess.)

Reply:
I see, I think this is fine then. Nice to have the indicator in the binlog when digging deep but not something that needs to be shown during every build. Thanks.

Side note, since source-build *needs* to do this without other platforms' output, it disables the feature that makes this target run on other RIDs:

https://github.com/dotnet/core-setup/blob/ce971b238196b08e9423bf9a5bfd192b8f7e1e0b/src/pkg/packaging-tools/framework.dependency.targets#L148-L149

That solution is a gap in the artifact, so probably not something we can do in the Microsoft build without decent planning.